server/etcdserver/server.go

```diff
@@ -2082,6 +2090,9 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 			}
 			lg.Panic("failed to create snapshot", zap.Error(err))
 		}
+
+		verifyConsistentIndexIsLatest(lg, snap, s.consistIndex.ConsistentIndex())
```

Comment:
How does this new verify function call relate to the snippet below from earlier in the function?

```go
// commit kv to write metadata (for example: consistent index) to disk.
// This guarantees that Backend's consistent_index is >= index of last snapshot.
s.KV().Commit()
```

Are we saying that the commented guarantee is no longer accurate or is there some kind of timing issue meaning we need to reverify?

Reply:
This change is codifying/verifying the comment.

Reply:
Right - given the strong language in the comment I had assumed there was already some verification happening somewhere.  Thanks for raising this.